### PR TITLE
Feature/make fixed width datastore resource

### LINKED ISSUE #1343: Allow fixed width on HDFS

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
Fixes #1343 
The fixed width datastore can take a resource.

### CODE REVIEW

File: engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java [hunk 1]
```diff
@@ -22,154 +22,215 @@
...
-		return new PerformanceCharacteristicsImpl(false, true);
-	}
-
-	@Override
-	protected UsageAwareDatastoreConnection<DataContext> createDatastoreConnection() {
-		final File file = new File(_filename);
-		assert file.exists();
-		final FixedWidthConfiguration configuration;
-		
-		if (_fixedValueWidth == -1) {
```
Comment (ClaudiaPHI) [inline engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java]: This line

File: engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java [hunk 2]
```diff
@@ -22,154 +22,227 @@
...
+    }
+
+    public FixedWidthDatastore(String name, Resource resource, String filename, String encoding, int[] valueWidths,
+            boolean failOnInconsistencies, boolean skipEbcdicHeader, boolean eolPresent, int headerLineNumber) {
+        super(name);
+        _filename = filename;
+        if (resource == null) {
+            resource = new FileResource(filename);
+        }
+        _resourceRef = new SerializableRef<Resource>(resource);
```
Comment (jhorcicka) [inline engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java]: I would remove "Resource".

File: engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java [hunk 3]
```diff
@@ -22,154 +22,227 @@
...
+
+    private void readObject(ObjectInputStream stream) throws IOException, ClassNotFoundException {
+        final Adaptor adaptor = new Adaptor() {
+            @Override
+            public void deserialize(GetField getField, Serializable serializable) throws Exception {
+                final String filename = (String) getField.get("_filename", "");
+                final Field field = FixedWidthDatastore.class.getDeclaredField("_resourceRef");
+                field.setAccessible(true);
+                final FileResource fileResource = new FileResource(filename);
+                final SerializableRef<Resource> resourceRef = new SerializableRef<Resource>(fileResource);
```
Comment (jhorcicka) [inline engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java]: And here.

File: engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java [hunk 4]
```diff
@@ -22,154 +22,227 @@
...
+    }
+
+    public FixedWidthDatastore(String name, Resource resource, String filename, String encoding, int fixedValueWidth,
+            boolean failOnInconsistencies, boolean skipEbcdicHeader, boolean eolPresent, int headerLineNumber) {
+        super(name);
+        _filename = filename;
+        if (resource == null) {
+            resource = new FileResource(filename);
+        }
+        _resourceRef = new SerializableRef<Resource>(resource);
```
Comment (jhorcicka) [inline engine/core/src/main/java/org/datacleaner/connection/FixedWidthDatastore.java]: And here.

**Comment by LosD** [comment]
Why is this still reformatting entire files?

**Comment by ClaudiaPHI** [comment]
It reformatts because classes have changes designation.  It fails because of metamodel snaphot dependency.

**Comment by ClaudiaPHI** [comment]
Ready for full testing if anyone wants to try, otherwise merge it.

**Comment by LosD** [comment]
I think I pointed out more than once that we do not do full file reformats.

And we do not merge failing code

**Comment by ClaudiaPHI** [comment]
I cannot do anything with class reformats if github thinks it's a different class because I changed the class that is being implemented.

**Comment by LosD** [comment]
Hmmm... I'm not sure I understand? The problem, as I see it, is that FixedWidthDatastoreDialog has been changed from tabs to spaces. That is fine, just not in the context of a bug.

**Comment by jhorcicka** [comment]
FixedWidthDatastore and FixedWidthDatastoreDialog indentation was reverted to tabs.

**Comment by ClaudiaPHI** [comment]
I am using the eclipse formatter from DataCleaner. I can't do anything to the identation. Please merge it as it is. The tabs indentation is old.

**Comment by jhorcicka** [comment]
@LosD, can you take a final look at this? I believe that from my and @ClaudiaPHI's point of view it's ready for merging.

**Pull request merged**